pkg/cmd/delete.go

```diff
@@ -120,3 +127,53 @@ func (command *deleteCmdOptions) run(args []string) error {
 
 	return nil
 }
+
+func getIntegration(ctx context.Context, c client.Client, name string, namespace string) (*v1.Integration, error) {
+	key := k8sclient.ObjectKey{
+		Name:      name,
+		Namespace: namespace,
+	}
+	answer := v1.NewIntegration(namespace, name)
+	if err := c.Get(ctx, key, &answer); err != nil {
+		return nil, err
+	}
+	return &answer, nil
+}
+
+func deleteIntegration(ctx context.Context, c client.Client, integration *v1.Integration) error {
+	deleted, binding, err := deleteKameletBindingIfExists(ctx, c, integration)
+	if err != nil {
+		return err
+	}
+	if deleted {
+		// Deleting KameletBinding will automatically clean up the integration
+		fmt.Println("KameletBinding " + binding + " deleted")
+		return nil
+	}
+	return c.Delete(ctx, integration)
```

Comment:
Can you skip this when the KameletBinding is deleted? Deletion of integration is done in cascade, otherwise there might be temporary recreations.

Reply:
Actually, the code already relies on the cascading delete. In this code, when the KameletBinding is `deleted` it early returns in the previous block so `c.Delete(ctx, integration)` won't be executed. Isn't it what you suggest?